File: Alarmservice/app/src/main/java/com/daeseong/alarmservice/AlarmReceiver.java
package com.daeseong.alarmservice;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.util.Log;

public class AlarmReceiver extends BroadcastReceiver {

    private static final String TAG = AlarmReceiver.class.getSimpleName();

    @Override
    public void onReceive(Context context, Intent intent) {

        try {

            String val = intent.getExtras().getString("alarm");
            Intent serviceIntent  = new Intent(context, AlarmService.class);
            serviceIntent .putExtra("alarm", val);
            startServiceCompat(context, serviceIntent);

        } catch (Exception ex) {
            Log.e(TAG, ex.getMessage().toString());
        }
    }

    private void startServiceCompat(Context context, Intent intent) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(intent);
        } else {
            context.startService(intent);
        }
    }
}

File: alarmservice(kotlin)/app/src/main/java/com/daeseong/alarmservice/AlarmReceiver.kt
package com.daeseong.alarmservice

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.util.Log

class AlarmReceiver : BroadcastReceiver() {

    private val tag = AlarmReceiver::class.java.simpleName

    override fun onReceive(context: Context, intent: Intent) {

        try {
            val value = intent.extras?.getString("alarm")
            val serviceIntent = Intent(context, AlarmService::class.java)
            serviceIntent.putExtra("alarm", value)
            startServiceCompat(context, serviceIntent)
        } catch (ex: Exception) {
            Log.e(tag, ex.message.toString())
        }
    }

    private fun startServiceCompat(context: Context, intent: Intent) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            context.startForegroundService(intent)
        } else {
            context.startService(intent)
        }
    }
}
